lib/make-disk-image.nix

```diff
@@ -55,7 +57,7 @@ let
 
     # populate nix db, so nixos-install doesn't complain
     export NIX_STATE_DIR=$TMPDIR/state
-    nix-store --load-db < ${pkgs.closureInfo {
+    ${pkgs.fakeroot}/bin/fakeroot nix-store --register-validity --reregister < ${closureInfo {
```

Comment:
This is a bit slower but solves a real issue as it needs to recompute all checksums.
If the performance overhead is not acceptable, I can also hide this behind a conditional.

Reply:
I decided to drop this approach again.